modules/repository-s3/src/javaRestTest/java/org/elasticsearch/repositories/s3/RepositoryS3RestReloadCredentialsIT.java

```diff
@@ -77,15 +75,21 @@ public void testReloadCredentialsFromKeystore() throws IOException {
         final var accessKey1 = randomIdentifier();
         repositoryAccessKey = accessKey1;
         keystoreSettings.put("s3.client.default.access_key", accessKey1);
-        keystoreSettings.put("s3.client.default.secret_key", randomIdentifier());
+        keystoreSettings.put("s3.client.default.secret_key", randomSecretKey());
         cluster.updateStoredSecureSettings();
-        assertOK(client().performRequest(new Request("POST", "/_nodes/reload_secure_settings")));
+        final Request reloadSecureSettingsRequest = new Request("POST", "/_nodes/reload_secure_settings");
+        reloadSecureSettingsRequest.setJsonEntity("""
+            {
+              "secure_settings_password": "keystore-password"
+            }
+            """);
```

Comment:
Slight preference for `newXContentRequest` here rather than forcing JSON with a literal body:

```suggestion
        final Request reloadSecureSettingsRequest = newXContentRequest(
            HttpMethod.POST,
            "/_nodes/reload_secure_settings",
            (b, p) -> b.field("secure_settings_password", "keystore-password")
        );
```

Reply:
I was not aware of this helper method. Thanks!